### DISCUSSION TITLE
Can we add a progress report in activity

### DISCUSSION THREAD

**Discussion by dust63** [elsa-workflows/elsa-core#600] [Ideas]
Can you think we can add some mechanism on activity to report a progress state. 
In example:
An activity that copy file.
An activity that making a video encoding.
...

**Comment by sfmskywalker** [comment]
Maybe. But I'm thinking it might make more sense to have a "CopyFile" activity that works as follows:

1. Kicks off a separate process that copies the file (perhaps using the service bus or some other background job).
2. The activity suspends the workflow (workflow gets persisted and is removed from memory).
3. Once the copying process is finished, the workflow is resumed.

Reporting progress would be done from the background job.
Would this work for your scenario, or is there a specific reason progress state needs to be reported form the workflow? If so, can you elaborate a little bit more on the use case?
